src/hooks/useSelectionContainer.tsx

```diff
@@ -51,8 +51,7 @@ export function useSelectionContainer<T extends HTMLElement>({
     eventsElement,
   });
 
-  const DragSelection = () => <MouseSelection ref={containerRef} {...selectionProps} />;
-  DragSelection.displayName = 'DragSelection';
+  const DragSelection = useCallback(() => <MouseSelection ref={containerRef} {...selectionProps} />, []);
```

Comment:
Because this wasn't memoized, whenever the parent component would re-render (like in our example), it would create a new version of `<DragSelection />` which was blowing away the old state and replacing it with a brand new instance